zkevm-circuits/src/evm_circuit/execution/calldatacopy.rs

```diff
@@ -191,8 +190,11 @@ impl<F: Field> ExecutionGadget<F> for CallDataCopyGadget<F> {
             ),
         )?;
         let src_id = if call.is_root { tx.id } else { call.caller_id };
-        self.src_id
-            .assign(region, offset, Some(F::from(src_id as u64)))?;
+        self.src_id.assign(
+            region,
+            offset,
+            Some(F::from(u64::try_from(src_id).unwrap())),
```

Comment:
Why not simply `F::from(src_id as u64)` ?

Reply:
I'm a big fan of not using `as` to convert values because it doesn't trigger some clippy warning. E.g.
```
let x: u64 = 9;
let _y: u64 = x as u64; // clippy is ok with this.
let _z: u64 = x.into(); // error: useless conversion to the same type: `u64`
let _a: u64 = x.try_into().unwrap(); // error: useless conversion to the same type: `u64`
```

Reply:
When I have the time, I want to implement `ToScalar` for `usize` and `u64` so we can all type less.